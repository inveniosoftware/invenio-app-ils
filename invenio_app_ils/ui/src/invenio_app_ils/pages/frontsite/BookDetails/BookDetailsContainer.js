import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { BookDetails } from './components';

export default class BookDetailsContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchBookDetails = this.props.fetchBookDetails;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.documentPid) {
        this.fetchBookDetails(location.state.documentPid);
      }
    });
    this.fetchBookDetails(this.props.match.params.documentPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    return (
      <Container>
        <BookDetails />
      </Container>
    );
  }
}

BookDetailsContainer.propTypes = {
  fetchBookDetails: PropTypes.func.isRequired,
};
