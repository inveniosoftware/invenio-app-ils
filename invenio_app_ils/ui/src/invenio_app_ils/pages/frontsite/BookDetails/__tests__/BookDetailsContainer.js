import React from 'react';
import { shallow, mount } from 'enzyme';
import BookDetailsContainer from '../BookDetailsContainer';
jest.mock('../components/BookDetails', () => {
  return {
    BookDetails: () => null,
  };
});

describe('BookDetailsContainer tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      documentPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <BookDetailsContainer
        history={routerHistory}
        match={routerUrlParams}
        fetchBookDetails={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch item details on mount', () => {
    const mockedFetchBookDetails = jest.fn();
    component = mount(
      <BookDetailsContainer
        history={routerHistory}
        match={routerUrlParams}
        fetchBookDetails={mockedFetchBookDetails}
      />
    );
    expect(mockedFetchBookDetails).toHaveBeenCalledWith(
      routerUrlParams.params.documentPid
    );
  });
});
