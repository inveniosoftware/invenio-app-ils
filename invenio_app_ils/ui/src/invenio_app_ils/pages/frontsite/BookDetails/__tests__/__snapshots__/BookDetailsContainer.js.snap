// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetailsContainer tests should load the details component 1`] = `
<Container>
  <BookDetails />
</Container>
`;
